### PR TITLE
Support Node.js

### PR DESCRIPTION
**Pull request by ferrell-code**
Support node

~~todo: Add documentation for using manta.js in node~~

Simple docs added to README

### CODE REVIEW

File: manta-js/package/package.json [hunk 1]
```diff
@@ -15,7 +16,10 @@
     "wasm-postbuild": "rm ./dist/wallet/crate/pkg/.gitignore",
     "wasm-dev": "wasm-pack build ./src/wallet/crate",
     "wasm": "wasm-pack build ./src/wallet/crate --out-dir ../../../dist/wallet/crate/pkg",
-    "build": "yarn prebuild && yarn wasm && yarn wasm-dev && tsc && yarn wasm-postbuild"
+    "build": "yarn prebuild && yarn wasm && yarn wasm-dev && tsc && yarn wasm-postbuild",
+    "wasm-dev-node": "wasm-pack build --target nodejs ./src/wallet/crate",
```
Comment (ferrell-code) [inline manta-js/package/package.json:20]: If we want to support node I think we would have to publish a node npm package as well, which is a bit annoying

Comment (ghzlatarev) [reply manta-js/package/package.json:19]: Can we support it from 1 branch or we'll need multiple branches as well?

Comment (ghzlatarev) [reply manta-js/package/package.json:19]: So are more changes needed or is this pretty much the final version ?

Comment (ferrell-code) [reply manta-js/package/package.json:19]: No this is it, other than writing docs. 

This should work without creating another branch. For browser run `yarn build`  and for node `yarn build-node`. This works for both for me, ran the example to (lightly) test for browser and tested the node build on local project https://github.com/ferrell-code/node-manta-example

Comment (ferrell-code) [reply manta-js/package/package.json:19]: added some simple docs in README

Comment (ghzlatarev) [reply manta-js/package/package.json:19]: Ok thanks, let me check out the node version with my test in manta.

Comment (ghzlatarev) [reply manta-js/package/package.json:19]: Nice, seems im finally hitting an internal SDK error:
```
Initial sync failed. Error during asynchronous call: SignerConnectionError(reqwest::Error { kind: Builder, source: "JsValue(ReferenceError: Headers is not defined\nReferenceError: Headers is not defined\n    at /home/georgi/Desktop/workspace/Manta/tests/node_modules/manta.js/dist/wallet/crate/pkg/manta_wasm_wallet.js:1219:17\n    at handleError (/home/georgi/Desktop/workspace/Manta/tests/node_modules/manta.js/dist/wallet/crate/pkg/manta_wasm_wallet.js:251:18)\n    at module.exports.__wbg_new_2d0053ee81e4dd2a (/home/georgi/Desktop/workspace/Manta/tests/node_modules/manta.js/dist/wallet/crate/pkg/manta_wasm_wallet.js:1218:65)\n    at wasm://wasm/0036a466:wasm-function[936]:0x9bfa8\n    at wasm://wasm/0036a466:wasm-function[64]:0x134e5\n    at wasm://wasm/0036a466:wasm-function[175]:0x56ed4\n    at wasm://wasm/0036a466:wasm-function[81]:0x248e4\n    at wasm://wasm/0036a466:wasm-function[184]:0x59e8a\n    at wasm://wasm/0036a466:wasm-function[182]:0x5933e\n    at wasm://wasm/0036a466:wasm-function[686]:0x942f3\n    at wasm://wasm/0036a466:wasm-function[411]:0x7fa62\n    at wasm://wasm/0036a466:wasm-function[1045]:0x9d39f\n    at __wbg_adapter_28 (/home/georgi/Desktop/workspace/Manta/tests/node_modules/manta.js/dist/wallet/crate/pkg/manta_wasm_wallet.js:210:10)\n    at real (/home/georgi/Desktop/workspace/Manta/tests/node_modules/manta.js/dist/wallet/crate/pkg/manta_wasm_wallet.js:195:20)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5))" })
```

Comment (ghzlatarev) [reply manta-js/package/package.json:19]: Tried it and it works.

**Review by ghzlatarev — DISMISSED** [review]

**Review by Kevingislason — COMMENTED** [review]
Nice work, I tried to figure this stuff out and it was a nightmare. 

After thinking it over, I think we may want to publish both the browser and node.js versions of the code together as a single npm package, assuming that doesn't break things. Users of node.js would simply import from `manta.js/node`, and users of manta.js would import from `manta.js/browser`. Running `yarn build` would run scripts to build for both targets. Since this would break current import paths, we would want to bump the major version of this package as well.

I think this change would help avoid human error while publishing these packages to npm, which could be potentially very serious. It's not strictly correct to have one package.json file for two different packages, because a package can only have one `name`, and that's what npm looks at when publishing. Although I would prefer to publish these packages in CI, we currently publish packages manually, for better or worse. Whenever I publish manta.js given the configuration in this PR, I would have to manually update the name in the package.json, and be very careful not to confuse the two different build targets. If the process for publishing packages isn't really straightforward there's room for human error, and I've definitely made mistakes in the past.

**Comment by ghzlatarev** [comment]
> Nice work, I tried to figure this stuff out and it was a nightmare.
> 
> After thinking it over, I think we may want to publish both the browser and node.js versions of the code together as a single npm package, assuming that doesn't break things. Users of node.js would simply import from `manta.js/node`, and users of manta.js would import from `manta.js/browser`. Running `yarn build` would run scripts to build for both targets. Since this would break current import paths, we would want to bump the major version of this package as well.
> 
> I think this change would help avoid human error while publishing these packages to npm, which could be potentially very serious. It's not strictly correct to have one package.json file for two different packages, because a package can only have one `name`, and that's what npm looks at when publishing. Although I would prefer to publish these packages in CI, we currently publish packages manually, for better or worse. Whenever I publish manta.js given the configuration in this PR, I would have to manually update the name in the package.json, and be very careful not to confuse the two different build targets. If the process for publishing packages isn't really straightforward there's room for human error, and I've definitely made mistakes in the past.

@ferrell-code

**Comment by ferrell-code** [comment]
Ok this should make it so manta.js defaults to browser build. Please double  check this doesn't break anything. It works in both node and browser environment for me

**Comment by ghzlatarev** [comment]
> Ok this should make it so manta.js defaults to browser build. Please double check this doesn't break anything. It works in both node and browser environment for me.
> 
> No changes to browser... I could make it more explicit exporting browser build to `./browser` instead of `./`, but this way it isn't a breaking change

So still build it with `yarn build-node` right ?

**Comment by ferrell-code** [comment]
Yeah or there is now a `build-all` to build both and what ultimately should be run when publishing npm package

**Comment by ferrell-code** [comment]
Also needed to add a few more specs to node specific README. Not sure how easy to follow along it is

**Comment by ghzlatarev** [comment]
> browser
Getting this again after building with `yarn build-all` and package.json line: `"manta.js": "file:../../Manta-Network/sdk/manta-js/package",`
```
TSError: ⨯ Unable to compile TypeScript:
rpc_full_sync_test.ts:2:58 - error TS2307: Cannot find module 'manta.js/node' or its corresponding type declarations.

2 import { MantaPrivateWallet, Environment, Network } from 'manta.js/node';
                                                           ~~~~~~~~~~~~~~~
```

**Comment by ghzlatarev** [comment]
> Also needed to add a few more specs to node specific README. Not sure how easy to follow along it is



> Also needed to add a few more specs to node specific README. Not sure how easy to follow along it is

Yeah we need to mention it here https://github.com/Manta-Network/sdk/blob/main/manta-js/README.md?plain=1#L13-L18

**Review by ghzlatarev — APPROVED** [review]

**Review by ghzlatarev — DISMISSED** [review]

**Review by Kevingislason — APPROVED** [review]
tested on front end, looks good

**Review by Kevingislason — CHANGES_REQUESTED** [review]
err sorry, one more thing, still need to bump the version. I think 0.1.0 makes sense since there is significant new functionality but changes are not breaking

**Review by ghzlatarev — APPROVED** [review]

**Review by Kevingislason — APPROVED** [review]

**Comment by Kevingislason** [comment]
Just lmk and I can publish the package

**Comment by ghzlatarev** [comment]
> Just lmk and I can publish the package

Merge and publish

**Comment by ferrell-code** [comment]
good on my end, we can bundle #71 in this new release too, or have it bump another version

**Comment by Kevingislason** [comment]
I'm gonna bump the version to 1.0.0, I think there was a previously published version 0.1.0, and npm won't let me publish over it

**Review by Kevingislason — APPROVED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**